bridges/modules/xcm-bridge-hub/src/lib.rs

```diff
@@ -56,63 +202,1513 @@ pub mod pallet {
+		type BridgeOriginAccountIdConverter: ConvertLocation<AccountIdOf<ThisChainOf<Self, I>>>;
+
+		/// Amount of this chain native tokens that is reserved on the sibling parachain account
+		/// when bridge open request is registered.
+		#[pallet::constant]
+		type BridgeDeposit: Get<BalanceOf<ThisChainOf<Self, I>>>;
+		/// Currency used to pay for bridge registration.
+		type Currency: MutateHold<
+			AccountIdOf<ThisChainOf<Self, I>>,
+			Balance = BalanceOf<ThisChainOf<Self, I>>,
+			Reason = Self::RuntimeHoldReason,
+		>;
+		/// The overarching runtime hold reason.
+		type RuntimeHoldReason: From<HoldReason<I>>;
+		/// Do not hold `Self::BridgeDeposit` for the location of `Self::OpenBridgeOrigin`.
+		/// For example, it is possible to make an exception for a system parachain or relay.
+		type AllowWithoutBridgeDeposit: Contains<Location>;
+
+		/// Local XCM channel manager.
+		type LocalXcmChannelManager: LocalXcmChannelManager;
+		/// XCM-level dispatcher for inbound bridge messages.
+		type BlobDispatcher: DispatchBlob;
 	}
 
+	/// An alias for the bridge metadata.
+	pub type BridgeOf<T, I> = Bridge<ThisChainOf<T, I>>;
+	/// An alias for this chain.
+	pub type ThisChainOf<T, I> =
+		pallet_bridge_messages::ThisChainOf<T, <T as Config<I>>::BridgeMessagesPalletInstance>;
+	/// An alias for the associated lanes manager.
+	pub type LanesManagerOf<T, I> =
+		pallet_bridge_messages::LanesManager<T, <T as Config<I>>::BridgeMessagesPalletInstance>;
+
 	#[pallet::pallet]
+	#[pallet::storage_version(migration::STORAGE_VERSION)]
 	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
 
 	#[pallet::hooks]
 	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
 		fn integrity_test() {
 			assert!(
-				Self::bridged_network_id().is_some(),
+				Self::bridged_network_id().is_ok(),
 				"Configured `T::BridgedNetwork`: {:?} does not contain `GlobalConsensus` junction with `NetworkId`",
 				T::BridgedNetwork::get()
 			)
 		}
+
+		#[cfg(feature = "try-runtime")]
+		fn try_state(_n: BlockNumberFor<T>) -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::do_try_state()
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Open a bridge between two locations.
+		///
+		/// The caller must be within the `T::OpenBridgeOrigin` filter (presumably: a sibling
+		/// parachain or a parent relay chain). The `bridge_destination_universal_location` must be
+		/// a destination within the consensus of the `T::BridgedNetwork` network.
+		///
+		/// The `BridgeDeposit` amount is reserved on the caller account. This deposit
+		/// is unreserved after bridge is closed.
+		///
+		/// The states after this call: bridge is `Opened`, outbound lane is `Opened`, inbound lane
+		/// is `Opened`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(Weight::zero())] // TODO:(bridges-v2) - https://github.com/paritytech/polkadot-sdk/pull/4949 - add benchmarks impl - FAIL-CI
+		pub fn open_bridge(
```

Comment:
I thought the bridge was between two global consensuses and then different parachains in these consensuses had lanes between them. But given these functions, these connections would be bridges? What's the model to think about lanes then?

Reply:
@franciscoaguirre 
BridgeHubs on both side manages lanes (basically, queues or structs which holds some metadata) and the lane can be created e.g. between two parachains from different consensus, the lane can be also between relaychains.

E.g.

Lane_01:  AssetHubPolkadot -> BridgeHubPolkadot <> BridgeHubKusama <- AssetHubKusama
Lane_02:  Acala -> BridgeHubPolkadot <> BridgeHubKusama <- Encointer
Lane_03:  Polkadot -> BridgeHubPolkadot <> BridgeHubKusama <- Encointer
Lane_04:  CollectivesPolkadot -> BridgeHubPolkadot <> BridgeHubKusama <- Kusama

Please, check the example in the docs: https://github.com/paritytech/polkadot-sdk/blob/b974795ba8984471cb76b254ee41e485c8d0ce0e/bridges/modules/xcm-bridge-hub/src/lib.rs#L114-L141
If it is still not clear or missing anything, let's fix the docs here.

Reply:
They are still called "bridges" as well as "lanes" and are opened with `fn open_bridge` instead of something like `fn open_lane`? That's what's a little confusing, nothing more

Reply:
I would say that, from the caller's perspective, it is a "bridge to another consensus somewhere." The pallet_xcm_bridge_hub stores Bridges (bridge metadata, such as who is connected with whom, deposits, and any other details needed to manage the bridge), while the lane is a hidden implementation detail. The lanes are stored in the pallet_bridge_messages pallet, and the caller doesn't need to know about the lanes or other specifics—only that someone needs to run a relayer for this lane.